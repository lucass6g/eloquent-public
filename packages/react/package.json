{
  "name": "@eloquent/react",
  "version": "1.0.0",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup",
    "clean": "rm -rf dist && rm -rf node_modules"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.1.0",
    "@types/react": "^18.3.3",
    "@types/react-dom": "^18.3.0",
    "react": "^18.3.1",
    "typescript": "^5.4.5"
  }
}
