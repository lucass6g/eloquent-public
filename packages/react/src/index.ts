export {Switch, Case, Default} from './components/conditional-switch'