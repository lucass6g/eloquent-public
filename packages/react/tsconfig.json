{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
