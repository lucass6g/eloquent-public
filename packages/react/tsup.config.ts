import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})