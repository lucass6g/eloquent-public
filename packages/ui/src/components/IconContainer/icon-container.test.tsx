import { render, screen } from '@testing-library/react';
import { ComponentIcon } from 'lucide-react';
import { describe, expect, it } from 'vitest';
import { Icon } from '.';


describe('Icon Container Story', () => {
    it('renders with default size', () => {
        const component = render(<Icon.Container data-testid="container" size='medium'>
            <ComponentIcon />
        </Icon.Container>)

        expect(screen.getByTestId('container')).toHaveClass('[&>svg]:size-6');
        expect(component.baseElement).toMatchSnapshot();
    });
    it('renders with small size', () => {
        render(<Icon.Container data-testid="container" size='small'>
            <ComponentIcon />
        </Icon.Container>)

        expect(screen.getByTestId('container')).toHaveClass('[&>svg]:size-4');
    });
    it('renders with large size', () => {
        render(<Icon.Container data-testid="container" size='large'>
            <ComponentIcon />
        </Icon.Container>)

        
        expect(screen.getByTestId('container')).toHaveClass('[&>svg]:size-8');
    });
});