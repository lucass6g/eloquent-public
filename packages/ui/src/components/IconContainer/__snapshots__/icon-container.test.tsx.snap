// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon Container Story > renders with default size 1`] = `
<body>
  <div>
    <div
      class="[&>svg]:size-6"
      data-testid="container"
    >
      <svg
        class="lucide lucide-component"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.5 8.5 9 12l-3.5 3.5L2 12l3.5-3.5Z"
        />
        <path
          d="m12 2 3.5 3.5L12 9 8.5 5.5 12 2Z"
        />
        <path
          d="M18.5 8.5 22 12l-3.5 3.5L15 12l3.5-3.5Z"
        />
        <path
          d="m12 15 3.5 3.5L12 22l-3.5-3.5L12 15Z"
        />
      </svg>
    </div>
  </div>
</body>
`;
